extensions/ffmpeg/src/main/java/com/google/android/exoplayer2/ext/ffmpeg/FfmpegDecoder.java

```diff
@@ -87,7 +88,17 @@ public FfmpegDecoderException decode(DecoderInputBuffer inputBuffer,
     }
     if (!hasOutputFormat) {
       channelCount = ffmpegGetChannelCount(nativeContext);
-      sampleRate = ffmpegGetSampleRate(nativeContext);
+
+      // Unlike the other decoders, ALAC does not set the sample rate, so we must get it ourselves.
+      // See https://trac.ffmpeg.org/ticket/6096
+      if ("alac".equals(codecName)) {
+        ParsableByteArray parsableExtraData = new ParsableByteArray(extraData);
+        parsableExtraData.setPosition(extraData.length - 4);
+        sampleRate = parsableExtraData.readUnsignedIntToInt();
+      } else {
+        sampleRate = ffmpegGetSampleRate(nativeContext);
```

Comment:
Yes, it currently returns 0. Looks like FFmpeg [zeroes out the context](https://github.com/FFmpeg/FFmpeg/blob/master/libavcodec/options.c#L95) so should be good. Tested with no issues.